vl-infotext.src.js

```diff
@@ -1,4 +1,11 @@
-import { NativeVlElement, define } from '/node_modules/vl-ui-core/vl-core.js';
+import { NativeVlElement, define, awaitScript, awaitUntil } from '/node_modules/vl-ui-core/vl-core.js';
+
+Promise.all([
+    awaitScript('util', '/node_modules/@govflanders/vl-ui-util/dist/js/util.min.js'),
+    awaitScript('infotext', '/node_modules/@govflanders/vl-ui-infotext/dist/js/infotext.min.js'),
+    awaitUntil(() => window.vl && window.vl.infotext)]
+  ).then(() => define('vl-infotext', VlInfotext, {extends: 'div'}));
```

Comment:
Als je dit vervangt met 

```
import '/node_modules/@govflanders/vl-ui-util/dist/js/util.min.js';
import '/node_modules/@govflanders/vl-ui-infotext/dist/js/infotext.min.js';
```

en dan onderaan de file `define('vl-infotext', VlInfotext, {extends: 'div'});` zet, heb je volgens mij het equivalente buiten de `awaitUntil(() => window.vl && window.vl.infotext)`. Dat laatste lijkt me echter niet nodig te zijn aangezien die vl.infotext synchroon gedefinieerd lijkt te worden.

Enkel `import('foobar.js')` is async (geeft een Promise terug), maar `import 'foobar.js'` zou sync moeten zijn.
Sources: 
> What happens with the ES6 module, according to the ECMAScript standard, is a very different set of steps than what is implemented in the CommonJS case. The first step, loading the contents of the file from disk is largely the same but may happen asynchronously. When the contents of the file are available, they are parsed. While parsing, the shape of the module as defined by the export statements is determined prior to evaluating the code. Once the shape is determined, the code is then evaluated. **It’s important to keep in mind that all import and export statements are resolved to their targets before any code is actually evaluated.** It is also important to note that the ES6 specification allows this resolution step to occur asynchronously. In Node.js terms, this means loading the contents of the script, resolution of the module imports and exports, and evaluation of the module code would occur over multiple turns of the event loop.
\- https://medium.com/the-node-js-collection/an-update-on-es6-modules-in-node-js-42c958b890c

> ECMAScript 6 gives you the best of both worlds: The synchronous syntax of Node.js plus the asynchronous loading of AMD. To make both possible, ES6 modules are syntactically less flexible than Node.js modules: Imports and exports must happen at the top level. That means that they can’t be conditional, either. **This restriction allows an ES6 module loader to analyze statically what modules are imported by a module and load them before executing its body.**
\- https://exploringjs.com/es6/ch_modules.html#_modules

Reply:
Ik heb dit offline besproken met @SanderKleykens. We hebben beslist om de code as is voorlopig te houden. Nadeel van de voorgestelde wijzigingen is dat libs (dus geen echte import/export modules) mogelijk meerdere keren zullen ingeladen worden. Dit kan potentieel voor performantie of override problemen zorgen.